modules/python/test/test_misc.py

```diff
@@ -50,39 +68,575 @@ class Arguments(NewOpenCVTests):
-        c = np.array([[[1,2],[3,4],[5,6]]], dtype='f')
+        c = np.array([[[1, 2], [3, 4], [5, 6]]], dtype='f')
         res4 = cv.utils.dumpInputArrayOfArrays([c, a, b])
         self.assertEqual(res4, "InputArrayOfArrays: empty()=false kind=0x00050000 flags=0x01050000 total(-1)=3 dims(-1)=1 size(-1)=3x1 type(0)=CV_32FC2 dims(0)=2 size(0)=3x1 type(0)=CV_32FC2")
 
+    def test_parse_to_bool(self):
+        for convertible in (True, False, 1, 0, 64, np.bool(1), np.uint8(0),
+                            np.int8(123), np.int16(11), np.int32(2),
+                            np.int64(1)):
+            expected = 'Bool: {}'.format(bool(convertible)).lower()
+            try:
+                actual = cv.utils.dumpBool(convertible).lower()
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, convertible,
+                        type(convertible).__name__
+                    )
+                )
+            else:
+                self.assertEqual(expected, actual,
+                                 msg="Can't convert {0} of type {1}\n"
+                                 "Expected: {2} vs Actual: {3}".format(
+                                     convertible,
+                                     type(convertible).__name__,
+                                     expected, actual
+                                 ))
+
+        a = np.array([1, 2, 0], dtype=np.bool)
+        for i in range(len(a)):
+            expected = 'Bool: {}'.format(bool(a[i])).lower()
+            try:
+                actual = cv.utils.dumpBool(a[i]).lower()
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, a[i],
+                        type(a[i]).__name__
+                    )
+                )
+            else:
+                self.assertEqual(expected, actual,
+                                 msg="Can't convert array element {0} of type {1}\n"
+                                 "Expected: {2} vs Actual: {3}".format(
+                                     a[i],
+                                     type(a[i]).__name__,
+                                     expected, actual
+                                 ))
+
+        with self.assertRaises(
+            TypeError,
+            msg='Exception is not risen for tuple slice with 1 element'
+        ):
+            _ = cv.utils.dumpBool((True, False, False)[0:1])
+
+        for not_convertible in (1.2, np.float(2.3), 's', 'str', (1, 2),
+                                [1, 2], a, 1 + 1j, 2j, np.complex(1.1),
+                                np.complex(-0.5, 1.), np.complex(imag=3j)):
```

Comment:
> np.complex(imag=3j)

I believe it should throw itself.
Please remove this case - there is already test for `<type 'complex'>`.

Reply:
I removed numpy complex alias.

`complex(imag=3j) == complex(-3)`